lib/utils/Utils.dart

```diff
@@ -1197,4 +1198,17 @@ class Utils {
     if (placemarks.isEmpty) { return null; }
     return placemarks.first.locality;
   }
+
+  static void requestInAppReview() async {
+    var myReports = await ApiSingleton().getReportsList();
+    var numReports = myReports.length;
+
+    if (numReports % 3 == 0 || numReports == 4){
```

Comment:
> This would introduce a bug, since numReports = 0 would trigger this if statement

I assumed that since this method is executed after a report is created, then the number would always be 1 or higher. But yes, it's always better to check and be sure. In this case, if I return when the list is smaller than 3, the bug is gone too. Does it make sense to use the "lastReviewRequest > 2 weeks" if the numReports is less than 3? (If someone creates reports, but then they delete them)

```
final prefs = await SharedPreferences.getInstance();
final lastReviewRequest = prefs.getInt(lastReviewRequestDatetime) ?? 0;
final lastReportCount = prefs.getInt(lastReviewRequestReportCountKey) ?? 0;
```
It would make sense to do that in UserManager.dart to keep consistency with the previous code

```
if (numReports == 3 || numReports == 4) {
  shouldRequestReview = true;
} else if (numReports >= lastReportCount + 3 || now - lastReviewRequest >= 14 * 24 * 60 * 60 * 1000) {
  shouldRequestReview = true;
}
```
Maybe it makes it more logic to put in the same `if` the 3 conditionals related to `numReports` and in the second one the one related with time, but that's not super important

Reply:
> Does it ever make sense to use the "lastReviewRequest > 2 weeks" if the numReports is less than 3? (If someone creates reports, but then they delete them)

Only once numReports is greater than 3